### PR TITLE
Fix error message when calc_* is set before structure

### PR DESCRIPTION
**Pull request by jan-janssen**
closes https://github.com/pyiron/pyiron/issues/665

### CODE REVIEW

**Comment by jan-janssen** [comment]
I am open for suggestions, otherwise I merge the changes now, because we should anyway define more clear rules when to raise warnings and when to throw errors. As by default the user does not see the warnings.

File: pyiron/lammps/base.py [hunk 1]
```diff
@@ -659,7 +659,7 @@ def calc_md(
             delta_temp=delta_temp,
             delta_press=delta_press,
             job_name=self.job_name,
-            rotation_matrix=self._prism.R
+            rotation_matrix=rotation_matrix
```
Comment (liamhuber) [inline pyiron/lammps/base.py:662]: nit: pep8 spaces around =

File: pyiron/lammps/base.py [hunk 2]
```diff
@@ -1146,7 +1146,8 @@ def _set_selective_dynamics(self):
                             "velocity___constraintz"
                         ] = "set NULL NULL 0.0"
 
-    def _ensure_requested_cell_deformation_allowed(self, pressure):
+    @staticmethod
+    def _ensure_requested_cell_deformation_allowed(structure, pressure, prism=None):
```
Comment (liamhuber) [inline pyiron/lammps/base.py:1150]: I don't find this name makes sense for something that returns a structure. (I think I might be at fault here, I know I had my fingers on this corner of the code at some point!) Do you think `_modify_structure_to_allow_requested_deformation` is better? I mean they're both really verbose, but if we're going to be verbose anyhow we might as well be precise :p

File: pyiron/lammps/base.py [hunk 3]
```diff
@@ -1160,18 +1161,48 @@ def _ensure_requested_cell_deformation_allowed(self, pressure):
...
+                        return skew_structure
                 except AttributeError:
                     warnings.warn(
                         "WARNING: Setting a calculation type which uses pressure before setting the structure risks " +
                         "constraining your cell shape evolution if non-diagonal pressures are used but the structure " +
                         "is not triclinic from the start of the calculation."
                     )
+        return structure
+
+    def _validate_structure(self, pressure):
```
Comment (liamhuber) [inline pyiron/lammps/base.py:1181]: This was actually the guy that got me thinking about names. I find it difficult to digest that `_validate_structure` returns `rotation_matrix`.

File: pyiron/lammps/base.py (hunk 1 above)
Comment (jan-janssen) [reply pyiron/lammps/base.py:662]: It is a function call, so: 
`PEP 8: E251 unexpected spaces around keyword / parameter equals`

**Pull request merged**